cli/audit.go

```diff
@@ -1,20 +1,362 @@
 package cli
 
 import (
+	"errors"
 	"fmt"
+	"path"
 	"strings"
 	"time"
 
-	log "github.com/Sirupsen/logrus"
+	logrus "github.com/Sirupsen/logrus"
 	"github.com/bitrise-io/go-utils/colorstring"
 	"github.com/bitrise-io/go-utils/command/git"
+	"github.com/bitrise-io/go-utils/fileutil"
+	"github.com/bitrise-io/go-utils/log"
 	"github.com/bitrise-io/go-utils/pathutil"
 	"github.com/bitrise-io/go-utils/retry"
 	"github.com/bitrise-io/stepman/models"
 	"github.com/bitrise-io/stepman/stepman"
+	"github.com/bitrise-io/stepman/validate"
 	"github.com/urfave/cli"
+	yaml "gopkg.in/yaml.v2"
 )
 
+var auditStepCommand = cli.Command{
+	Name:  "audit-step",
+	Usage: "Validates a Step and Step Share Params (id, version, uri).",
+	Flags: []cli.Flag{
+		cli.StringFlag{
+			Name:  "uri",
+			Usage: "The Step repository's git URI.",
+		},
+		cli.StringFlag{
+			Name:  "id",
+			Usage: "The Step ID.",
+		},
+		cli.StringFlag{
+			Name:  "version",
+			Usage: "The Step version.",
+		},
+		cli.StringFlag{
+			Name:  "dir",
+			Usage: "The Step's directory",
+		},
+	},
+	Action: func(c *cli.Context) error {
+		if err := auditStep(c); err != nil {
+			logrus.Fatal(err)
+		}
+		return nil
+	},
+}
+
+func auditStep(c *cli.Context) error {
+	uri := c.String("uri")
+	id := c.String("id")
+	version := c.String("version")
+	dir := c.String("dir")
+
+	fmt.Println()
+	log.Infof("Audting Step:")
+	log.Printf("- Step repository uri: %s", uri)
+	log.Printf("- Step id: %s", id)
+	log.Printf("- Step version: %s", version)
+	log.Printf("- Step dir: %s", dir)
+	fmt.Println()
+
+	if uri == "" {
```

Comment:
why don't you use `validate.StepParams` here too?

like it is used here https://github.com/bitrise-io/stepman/pull/236/files#diff-6cb035d629640ba91e8639a45e2ac243R60